File: benchmarks/suite5/scan.c
#include <stdint.h>
#include <stdio.h>
#include "bench.h"

int main(void) {
  const int64_t n = 8000000;
  int64_t acc = 0;
  for (int64_t i = 0; i < n; i++) {
    if (i % 3 == 0 || i % 7 == 0) {
      acc++;
    }
  }
  suite5_print_i64(acc);
  return 0;
}

File: benchmarks/suite5/scan.zig
const bench = @import("bench.zig");
const clib = @cImport({
    @cInclude("stdio.h");
});

pub fn main() void {
    const n: i64 = bench.opaqueI64(8000000);
    var acc: i64 = 0;
    var i: i64 = 0;
    while (i < n) : (i += 1) {
        if (@rem(i, 3) == 0 or @rem(i, 7) == 0) acc += 1;
    }
    bench.printI64(acc);
}
